src/NzbDrone.Core/IndexerSearch/EpisodeSearchService.cs

```diff
@@ -79,36 +114,33 @@ public void Execute(EpisodeSearchCommand message)
-                                                     {
-                                                         Page = 1,
-                                                         PageSize = 100000,
-                                                         SortDirection = SortDirection.Ascending,
-                                                         SortKey = "Id",
-                                                         FilterExpression = v => v.Monitored == true && v.Series.Monitored == true
-                                                     }).Records.ToList();
+            List<Episode> episodes;
 
-            var missing = episodes.Where(e => !_queueService.GetQueue().Select(q => q.Episode.Id).Contains(e.Id)).ToList();
-
-            _logger.ProgressInfo("Performing missing search for {0} episodes", missing.Count);
-            var downloadedCount = 0;
+            if (message.SeriesId > 0)
+            {
+                episodes = _episodeService.GetEpisodeBySeries(message.SeriesId)
+                                          .Where(e => e.Monitored && !e.HasFile)
+                                          .ToList();
+            }
 
-            //Limit requests to indexers at 100 per minute
-            using (var rateGate = new RateGate(100, TimeSpan.FromSeconds(60)))
+            else
             {
-                foreach (var episode in missing)
-                {
-                    rateGate.WaitToProceed();
-                    var decisions = _nzbSearchService.EpisodeSearch(episode);
-                    var processed = _processDownloadDecisions.ProcessDecisions(decisions);
-                    downloadedCount += processed.Grabbed.Count;
-                }
+                episodes = _episodeService.EpisodesWithoutFiles(new PagingSpec<Episode>
+                                                                    {
+                                                                        Page = 1,
+                                                                        PageSize = 100000,
+                                                                        SortDirection = SortDirection.Ascending,
+                                                                        SortKey = "Id",
+                                                                        FilterExpression =
+                                                                            v =>
+                                                                            v.Monitored == true &&
+                                                                            v.Series.Monitored == true
+                                                                    }).Records.ToList();
             }
 
-            _logger.ProgressInfo("Completed missing search for {0} episodes. {1} reports downloaded.", missing.Count, downloadedCount);
+            var missing = episodes.Where(e => !_queueService.GetQueue().Select(q => q.Episode.Id).Contains(e.Id)).ToList();
```

Comment:
get the queue outside and project it to episode Ids, otherwise we grab the queue once per item in the episode list and process the whole list multiple times.